src/wp-admin/options-media.php

```diff
@@ -44,6 +44,17 @@
 
 require_once ABSPATH . 'wp-admin/admin-header.php';
 
+/**
+ * Get upload organization preference.
+ *
+ * @since CP-2.1.0
+ *
+ * New option based on year only.
+ */
+$storefolders = '0';
+if ( ! empty( get_option( 'uploads_use_yearmonth_folders' ) ) ) {
```

Comment:
I wonder if this could be better handled by a database update with the release of a new version, update the setting if empty to contain '0'.

It would remove the need for an eternal `if` statement here if the upgrade is done in the same merge.

Reply:
I think that using the `$default_value` parameter of `get_option` is easier than updating the database, where the value can be already set.

Reply:
Normally I'd agree, but the current values are 1 or an empty string. This PR changes the option selector from a checkbox to a radio so the values are now '0', '1' and currently 'year', so handling before and after is not easy. There is a value so `get_option` will never return `false` for the default value to be used.

Reply:
The reason I've done it this way is to ensure compatibility with WP. Using '0' as the value in the radio works for WP, where the check is simply whether it is '1' or not. And it works in CP this way too because `empty` includes both a literal empty string and '0'.

So someone migrating from WP will have the radio button work immediately, and someone migrating back to WP will see the checkbox work. Changing the database won't be compatible.

Reply:
Some level of migration needs to be done when converting the input from checkbox to radio. I was wondering if we can do this in a database upgrade with a version release so the check becomes unnecessary.

Reply:
What about casting to `int` the option?
Both `(int)''` and `(int)'0'` casts to `0`.

Reply:
Testing locally I think that might work if 'year' can change to 2 and the checked() call on empty is also changed.

Reply:
If you cast both the call to get_option checked will work with 0.

Reply:
I'll submit a revised PR in a few hours.